File: source/tree/splaytree/java/SplayTreeTest.java

/**
 * Java 语言: 伸展树
 *
 * @author skywang
 * @date 2014/02/03
 */
public class SplayTreeTest {

	private static final int arr[] = {10,50,40,30,20,60};

	public static void main(String[] args) {
		int i, ilen;
		SplayTree<Integer> tree=new SplayTree<Integer>();

		System.out.print("== 依次添加: ");
		ilen = arr.length;
		for(i=0; i<ilen; i++) {
			System.out.print(arr[i]+" ");
			tree.insert(arr[i]);
		}

		System.out.print("\n== 前序遍历: ");
		tree.preOrder();

		System.out.print("\n== 中序遍历: ");
		tree.inOrder();

		System.out.print("\n== 后序遍历: ");
		tree.postOrder();
		System.out.println();

		System.out.println("== 最小值: "+ tree.minimum());
		System.out.println("== 最大值: "+ tree.maximum());
		System.out.println("== 树的详细信息: ");
		tree.print();

		i = 30;
		System.out.printf("\n== 旋转节点(%d)为根节点\n", i);
		tree.splay(i);
		System.out.printf("== 树的详细信息: \n");
		tree.print();

		// 销毁二叉树
		tree.clear();
	}
}

File: source/tree/splaytree/c/splaytree_test.c
/**
 * C 语言: 伸展树测试程序
 *
 * @author skywang
 * @date 2014/02/03
 */

#include <stdio.h>
#include "splay_tree.h"

static int arr[]= {10,50,40,30,20,60};
#define TBL_SIZE(a) ( (sizeof(a)) / (sizeof(a[0])) )

void main()
{
	int i, ilen;
	SplayTree root=NULL;

	printf("== 依次添加: ");
	ilen = TBL_SIZE(arr);
	for(i=0; i<ilen; i++)
	{
		printf("%d ", arr[i]);
		root = insert_splaytree(root, arr[i]);
	}

	printf("\n== 前序遍历: ");
	preorder_splaytree(root);

	printf("\n== 中序遍历: ");
	inorder_splaytree(root);

	printf("\n== 后序遍历: ");
	postorder_splaytree(root);
	printf("\n");

	printf("== 最小值: %d\n", splaytree_minimum(root)->key);
	printf("== 最大值: %d\n", splaytree_maximum(root)->key);
	printf("== 树的详细信息: \n");
	print_splaytree(root, root->key, 0);

	i = 30;
	printf("\n== 旋转节点(%d)为根节点\n", i);
	printf("== 树的详细信息: \n");
	root = splaytree_splay(root, i);
	print_splaytree(root, root->key, 0);

	// 销毁伸展树
	destroy_splaytree(root);
}
